server/src/resolvers/MemberResolver.ts

```diff
@@ -32,7 +32,16 @@ export default class MemberResolver {
 		return user;
 	}
 	@Mutation(() => Member)
-	async signUp(@Args() { username, email, password }: SignUpArgs): Promise<Member> {
+	async signUp(
+		@Args() { username, email, password, confirmedPassword }: SignUpArgs
+	): Promise<Member> {
+		let existingEmail = (await MemberServices.findOneBy({ email })) as Member;
+		let existingUsername = (await MemberServices.findOneBy({ username })) as Member;
+		if (existingEmail) throw Error(ErrorMessages.EMAIL_ALREADY_REGISTERED_ERROR_MESSAGE);
+		if (existingUsername) throw Error(ErrorMessages.USERNAME_ALREADY_REGISTERED_ERROR_MESSAGE);
```

Comment:
why are we using Error to send a message to client?

Reply:
Because it's important to check this in the server, and we need to throw an error an to stop the function and to inform cleint that there is an error... im not sure to understand your question

Reply:
Maybe it already doest it automatically with typeORM method? I'm not sure